### PR TITLE
Fix for hasEmptyAccount

### PR DESCRIPTION
**Pull request by smartcontracts**
## Description

Addresses https://github.com/ethereum-optimism/roadmap/issues/369. I also made sure that none of the other functions in the OVM state manager are impacted by similar issues.

### Fixes
- Fixes https://github.com/ethereum-optimism/roadmap/issues/369

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](./CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: core/vm/ovm_state_manager.go [hunk 1]
```diff
@@ -210,6 +210,20 @@ func testAndSetContractStorage(evm *EVM, contract *Contract, args map[string]int
 	return []interface{}{true}, nil
 }
 
+func hasEmptyAccount(evm *EVM, contract *Contract, args map[string]interface{}) ([]interface{}, error) {
```
Comment (smartcontracts) [inline core/vm/ovm_state_manager.go:213]: Original contract function is here for reference:

https://github.com/ethereum-optimism/contracts-v2/blob/7e82fce506df729dfd070e0205d98c1adc9e6dcd/contracts/optimistic-ethereum/OVM/execution/OVM_StateManager.sol#L185-L199

File: core/vm/ovm_state_manager.go [hunk 2]
```diff
@@ -210,6 +210,20 @@ func testAndSetContractStorage(evm *EVM, contract *Contract, args map[string]int
...
+	address, ok := args["_address"].(common.Address)
+	if !ok {
+		return nil, errors.New("Could not parse address arg in hasEmptyAccount")
+	}
+
+	contractHash := evm.StateDB.GetCodeHash(address)
```
Comment (tynes) [inline core/vm/ovm_state_manager.go:219]: Is it possible to have the nonce be 0 and the contract hash not be the empty code hash? That would sound like a bug to me if that ever happened

Comment (tynes) [reply core/vm/ovm_state_manager.go:219]: Just explaining out my though process here. The check for `contractHash != (common.Hash{})` is useful for when the account does not yet exist in the tree. If it is an EOA and has sent a single tx then the account will exist in the tree and the `emptyCodeHash` should be returned.

**Review by tynes — APPROVED** [review]

**Comment by smartcontracts** [comment]
FWIW this code is taken directly from the create function in evm.go.

**Comment by smartcontracts** [comment]
I made Karl aware that we will likely need to block users from setting their account nonce to zero.

**Pull request merged**